Sources/ApodiniDatabase/Query/QueryBuilder.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  File.swift
+//  
+//
+//  Created by Felix Desiderato on 08.12.20.
+//
+import Apodini
+import Foundation
+import Fluent
+import Vapor
+@_implementationOnly import Runtime
+
+internal struct QueryBuilder<Model: DatabaseModel> {
+    internal var debugDescription: String {
+        parameters.debugDescription
+    }
+    
+    private let type: Model.Type
+
+    private var fieldKeys: [FieldKey] {
+        type.keys
+    }
+
+    internal var parameters: [FieldKey: String] = [:]
+
+    init(type: Model.Type, queryString: String) {
+        self.type = type
+//        self.queryString = queryString
+        self.parameters = extract(from: queryString)
+    }
+    
+    init(type: Model.Type, parameters: [FieldKey: String]) {
+        self.type = type
+        self.parameters = parameters
+    }
+    
+    private func extract(from queryString: String) -> [FieldKey: String] {
+        var foundParameters: [FieldKey: String] = [:]
+        let extractedQueryString = queryString.split(separator: "?")[1]
```

Comment:
I would suggest to check the count or possible indexes before accessing a collection using a fixed index.